### PR TITLE
build: add option to disable installation of web assets

### LINKED ISSUE #4864: js files always installed
### What is the issue?

I've just updated the package for transmission in pkgsrc to 4.0.0.
We have separate packages for the daemon and the gtk and qt frontends. We want each installed file to belong only to one package.

I didn't find an option to stop CMake from installing the following files:
```
share/transmission/public_html/images/favicon.ico
share/transmission/public_html/images/favicon.png
share/transmission/public_html/images/webclip-icon.png
share/transmission/public_html/index.html
share/transmission/public_html/transmission-app.js
share/transmission/public_html/transmission-app.js.LICENSE.txt
```
I had assumed that `-DENABLE_WEB=OFF` would stop the installation, but it didn't.
Did I overlook an option, or is this a bug?

### Which application of Transmission?

transmission-daemon

### Which version of Transmission?

4.0.0

### PR DESCRIPTION
**Pull request by t-8ch**
Fixes #4864

### CODE REVIEW

**Review by ckerr — DISMISSED** [review]
LGTM.

Not merging since I want to give the primary CMake wrangler @mikedld time to look

**Comment by ckerr** [comment]
This has failed CI a handful of times, so maybe it actually is a PR issue. @t-8ch PTAL

File: CMakeLists.txt [hunk 1]
```diff
@@ -47,6 +47,7 @@ tr_auto_option(ENABLE_GTK "Build GTK client" AUTO)
 tr_auto_option(ENABLE_QT "Build Qt client" AUTO)
 tr_auto_option(ENABLE_MAC "Build Mac client" AUTO)
 tr_auto_option(REBUILD_WEB "Rebuild the web client's generated assets. Requires Node.js and network access." OFF)
+tr_auto_option(INSTALL_WEB "Install the web client's generated assets." ON)
```
Comment (mikedld) [inline CMakeLists.txt:50]: There's no `AUTO` handling (nor is it necessary).
```suggestion
option(INSTALL_WEB "Install the web client's generated assets." ON)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -778,7 +781,7 @@ foreach(P cli daemon gtk mac qt utils)
 endforeach()
 
 if(NOT REBUILD_WEB AND (ENABLE_DAEMON OR ENABLE_GTK OR ENABLE_QT))
-    tr_install_web(${CMAKE_INSTALL_DATAROOTDIR}/${TR_NAME})
+    tr_install_web(${CMAKE_SOURCE_DIR}/web/public_html)
```
Comment (mikedld) [inline CMakeLists.txt:784]: There's yet another `tr_install_web()` call in macosx/ that's now failing.

File: CMakeLists.txt [hunk 3]
```diff
@@ -507,6 +508,14 @@ if(ENABLE_UTP)
             -DLIBUTP_SHARED:BOOL=OFF)
 endif()
 
+function(tr_install_web SRC_DIR)
+    if (INSTALL_WEB)
```
[the remark this answers is not in the document]
Comment (t-8ch) [reply CMakeLists.txt:512]: Would it hurt to install it unconditionally?

[the remark this answers is not in the document]
Comment (mikedld) [reply CMakeLists.txt:512]: Not sure I understand the question, I thought you're adding `INSTALL_WEB` to do the exact opposite, i.e. not install it (meaning, the web client) unconditionally. What I'm saying is that if `INSTALL_WEB` is `OFF`, building the `pack-msi` target (defined in the dist/msi subdirectory) on Windows will fail, and the current approach is not to define that target if any of its dependencies are missing.

**Comment by t-8ch** [comment]
Care to take another look?

I missed the call to tr_install_web from the macosx folder, so the whole aproach was invalid.

**Review by mikedld — APPROVED** [review]

**Comment by ckerr** [comment]
@mikedld wdyt about this for `4.0.1`? I think this is safe but would like a 2nd opinion

**Comment by mikedld** [comment]
@ckerr, it fixes something for someone, right? :) Old behavior is preserved AFAICT, so shouldn't break anything new.

**Pull request merged**